aclk/legacy/aclk_query.c

```diff
@@ -237,6 +239,7 @@ int aclk_queue_query(char *topic, void *data, char *msg_id, char *query, int run
     new_query->data = data;
     new_query->next = NULL;
     new_query->created = now_realtime_usec();
+    now_realtime_timeval(&new_query->tv_in);
```

Comment:
If I am not mistaken one line above `new_query->created` can be calculated from this value as well thus avoiding calling `clock_gettime` twice (both `now_realtime_usec` and `now_realtime_timeval` call it internally). Also maybe the field `new_query->created` is obsoleted as it contains the same information just formated diferently

Reply:
as this is done while holding `ACLK_QUEUE_LOCK` that means we are blocking every Query Thread here thus reducing 2 syscalls to 1 makes sense here

Reply:
> If I am not mistaken one line above `new_query->created` can be calculated from this value as well thus avoiding calling `clock_gettime` twice (both `now_realtime_usec` and `now_realtime_timeval` call it internally). Also maybe the field `new_query->created` is obsoleted as it contains the same information just formated diferently

Looks like it can be calculated

`new_query->tv_in.tv_sec * USEC_PER_SEC + new_query->tv_in.tv_usec`